syntax = "proto2";

import "gate.proto";

package Record;

message  t_LoginRecord
{
	optional uint32 id = 1;
	optional uint32 type = 2;
}

//gate->record 请求角色信息
message t_Get_SelectInfo_GateRecord {
	optional uint32 accid = 1;
	optional string ip = 2;
}

//record->gate
message t_Ret_SelectInfo_GateRecord {
	optional uint32 accid = 1;
	optional string lastip = 2;
	optional uint32 offlinetime = 3;
	repeated .Gate.SelectUserInfo info = 4;
}

message t_CheckName_GateRecord {
	optional string name = 1;
	optional uint32 accid = 2;
}

message t_CheckName_Return_GateRecord {
	optional string name = 1;
	optional uint32 accid = 2;
	optional uint32 code = 3;
}

message t_CreateChar_GateRecord{
	optional uint32 accid = 1;
	optional string name = 2;
	optional uint32 country = 3;
	optional uint32 type = 4;
}

message t_CreateChar_Return_GateRecord {
	optional uint32 accid = 1;
	optional uint32 retcode = 2;
	optional .Gate.SelectUserInfo charinfo = 3;
}
